In [12]:
# import basic libraries 
import numpy as np
import pandas as pandas
import matplotlib.pyplot as plt
import seaborn as sb

%matplotlib inline

# load dataset into a pandas dataframe
df = pandas.read_csv('creditcardmarketing.csv')
df.head()

1  No  Air Miles    Letter    High  1.1 No.1  High.1  2  1.2  4 No.2  \
0  2  No  Air Miles    Letter  Medium    1   No  Medium  2    2  5  Yes   
1  3  No  Air Miles  Postcard    High    2   No  Medium  2    1  2  Yes   
2  4  No  Air Miles    Letter  Medium    2   No    High  1    1  4   No   
3  5  No  Air Miles    Letter  Medium    1   No  Medium  2    1  6  Yes   
4  6  No  Air Miles    Letter  Medium    1   No    High  3    1  4   No   

   1160.75    1669     877    1095   1002  
0   147.25    39.0   106.0    78.0  366.0  
1   276.50   367.0   352.0   145.0  242.0  
2  1219.00  1578.0  1760.0  1119.0  419.0  
3  1211.00  2140.0  1357.0   982.0  365.0  
4  1114.75  1847.0  1365.0   750.0  497.0

### Assess dataframe

In [13]:
# assess shape
df.shape

(17999, 17)

In [14]:
# assess column names
df.columns

Index(['1', 'No', 'Air Miles', 'Letter', 'High', '1.1', 'No.1', 'High.1', '2',
       '1.2', '4', 'No.2', '1160.75', '1669', '877', '1095', '1002'],
      dtype='object')

In [15]:
#data types of columns
df.dtypes

1              int64
No            object
Air Miles     object
Letter        object
High          object
1.1            int64
No.1          object
High.1        object
2              int64
1.2            int64
4              int64
No.2          object
1160.75      float64
1669         float64
877          float64
1095         float64
1002         float64
dtype: object

In [16]:
# list of columns, their non-null objects and data type of columns
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 17999 entries, 0 to 17998
Data columns (total 17 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   1          17999 non-null  int64  
 1   No         17999 non-null  object 
 2   Air Miles  17999 non-null  object 
 3   Letter     17999 non-null  object 
 4   High       17999 non-null  object 
 5   1.1        17999 non-null  int64  
 6   No.1       17999 non-null  object 
 7   High.1     17999 non-null  object 
 8   2          17999 non-null  int64  
 9   1.2        17999 non-null  int64  
 10  4          17999 non-null  int64  
 11  No.2       17999 non-null  object 
 12  1160.75    17975 non-null  float64
 13  1669       17975 non-null  float64
 14  877        17975 non-null  float64
 15  1095       17975 non-null  float64
 16  1002       17975 non-null  float64
dtypes: float64(5), int64(5), object(7)
memory usage: 2.3+ MB


In [17]:
df.head(5)

1  No  Air Miles    Letter    High  1.1 No.1  High.1  2  1.2  4 No.2  \
0  2  No  Air Miles    Letter  Medium    1   No  Medium  2    2  5  Yes   
1  3  No  Air Miles  Postcard    High    2   No  Medium  2    1  2  Yes   
2  4  No  Air Miles    Letter  Medium    2   No    High  1    1  4   No   
3  5  No  Air Miles    Letter  Medium    1   No  Medium  2    1  6  Yes   
4  6  No  Air Miles    Letter  Medium    1   No    High  3    1  4   No   

   1160.75    1669     877    1095   1002  
0   147.25    39.0   106.0    78.0  366.0  
1   276.50   367.0   352.0   145.0  242.0  
2  1219.00  1578.0  1760.0  1119.0  419.0  
3  1211.00  2140.0  1357.0   982.0  365.0  
4  1114.75  1847.0  1365.0   750.0  497.0

In [7]:
# id & air miles - do we need it?
# no ? no.1? no.2 ? need to rename columns
# 1.1, 2, 4 ? need to rename columns
# oh my god.. all the other numerical named columns freak me out lol :D 

**assumed column names**

1 - customer id

No - offer_accepted

Air Miles - air_miles

Letter - mailer_type

High - income_level

1.1 - bank_accounts_open

No.1 - overdraft_protection

High.1 - credit_rating

2 - credit_cards_held

1.2 - homes_owned

4 - household_size

No.2 - owns_home

1160.75 - avg_balance

1669 - balance

In [18]:
# means we have columns 877, 1095, 1002 which we are unsure what data they cover 
# shall we drop it?

# Summary of column data types assessment

- data types seem relevant to the columns
- rename columns to the suggestions
- drop column 1 and the last 3